stacks/build-stack.yaml

```diff
@@ -127,81 +176,136 @@ kubernetes:
+    apiVersion: apiextensions.k8s.io/v1beta1
+    kind: CustomResourceDefinition
     metadata:
-      name: nop
-      namespace: ${NAMESPACE}
+      name: taskruns.tekton.dev
     spec:
-      image: gcr.io/knative-releases/github.com/knative/build/cmd/nop@sha256:a318ee728d516ff732e2861c02ddf86197e52c6288049695781acb7710c841d4
+      additionalPrinterColumns:
+      - JSONPath: .status.conditions[?(@.type=="Succeeded")].status
+        name: Succeeded
+        type: string
+      - JSONPath: .status.conditions[?(@.type=="Succeeded")].reason
+        name: Reason
+        type: string
+      - JSONPath: .status.startTime
+        name: StartTime
+        type: date
+      - JSONPath: .status.completionTime
+        name: CompletionTime
+        type: date
+      group: tekton.dev
+      names:
+        categories:
+        - all
+        - tekton-pipelines
+        kind: TaskRun
+        plural: taskruns
+        shortNames:
+        - tr
+        - trs
+      scope: Namespaced
+      subresources:
+        status: {}
+      version: v1alpha1
     ---
-    apiVersion: build.knative.dev/v1alpha1
-    kind: ClusterBuildTemplate
+    apiVersion: tekton.dev/v1alpha1
+    kind: ClusterTask
     metadata:
       name: buildkit
     spec:
-      parameters:
-      - name: IMAGE
-        description: Where to publish the resulting image
-      - name: DOCKERFILE
-        description: The name of the Dockerfile
-        default: "Dockerfile"
-      - name: PUSH
-        description: Whether push or not
-        default: "true"
-      - name: DIRECTORY
-        description: The directory containing the app
-        default: "/workspace"
-      - name: BUILDKIT_CLIENT_IMAGE
-        description: The name of the BuildKit client (buildctl) image
-        default: "moby/buildkit:v0.3.1-rootless@sha256:2407cc7f24e154a7b699979c7ced886805cac67920169dcebcca9166493ee2b6"
-      - name: BUILDKIT_DAEMON_ADDRESS
-        description: The address of the BuildKit daemon (buildkitd) service
-        default: "tcp://buildkitd:1234"
+      inputs:
+        resources:
+        - name: source
+          type: git
+        params:
+        - name: image
+          description: Where to publish the resulting image
+        - name: dockerfile
+          description: The name of the Dockerfile
+        - name: docker-context
+          description: The context of the build
+        - name: push
+          description: Whether push or not
+          default: "true"
+        - name: workingdir
+          description: The directory containing the app
+          default: "/workspace"
+        - name: buildkit-client-image
+          description: The name of the BuildKit client (buildctl) image
+          default: "moby/buildkit:master"
+        - name: insecure-registry
+          description: Whether to use insecure registry
+      volumes:
+      - name: buildkit-cache
+        hostPath:
+          path: /var/lib/buildkit/runc-overlayfs/content
+          type: DirectoryOrCreate
```

Comment:
@tonistiigi Is it safe to share this directory across multiple BuildKit instances on the same host?